decode-ways/prograsshopper.py

```diff
@@ -0,0 +1,19 @@
+class Solution:
+    def numDecodings(self, s: str) -> int:
+        # sol 1: Time complexity O(n)
+        dp = {len(s): 1}
+
+        def dfs(start):
+            if start in dp:
+                return dp[start]
+            if s[start] == "0":
+                return 0
+            result = None
+            if start + 1 < len(s) and int(s[start:start+2]) < 27:
+                result = dfs(start+1) + dfs(start+2)
+            else:
+                result = dfs(start+1)
```

Comment:
요렇게 하시면 살짝 더 간단해질 수 있지 않을까 생각이 들었습니다. 물론 작성해주신 코드도 훌륭합니다! ㅎㅎ

```suggestion
            result = dfs(start + 1)
            if start + 1 < len(s) and 10 <= int(s[start:start+2]) < 27:
                result += dfs(start + 2)
```